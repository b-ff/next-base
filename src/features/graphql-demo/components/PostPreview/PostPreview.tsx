"use client";
import { Flex } from "@/components/Flex";
import { Post, User } from "@/schema";
import { formatPostDate } from "@/utils";
import Link from "next/link";
import { FC, HTMLProps, useMemo } from "react";

export type PostPreviewProps = HTMLProps<HTMLElement> & {
  post: Post;
  author: Partial<User>;
};

export const PostPreview: FC<PostPreviewProps> = ({ post, author }) => {
  const date = useMemo(
    () =>
      formatPostDate(
        Math.max(
          new Date(parseInt(post.createdAt))?.getTime(),
          new Date(parseInt(post.updatedAt))?.getTime()
        )
      ),
    [post.createdAt, post.updatedAt]
  );

  return (
    <Flex gap={2}>
      <Link href={`/posts/${post.id}`}>
        <h4>{post.title} &raquo;</h4>
      </Link>
      <small>
        Written by {author.name} at {date}
      </small>
    </Flex>
  );
};
