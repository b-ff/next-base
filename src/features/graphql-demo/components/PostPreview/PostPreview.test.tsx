import { PostPreview } from "./PostPreview";
import { render } from "@testing-library/react";

const mockPost = {
  id: 1,
  userId: "that-exact-user-id",
  title: "For whom the bell tolls?",
  body: "For those who refuse to take action and cross the boundaries. For those who complaining instead of doing things. For them we shall fire the torch and spread the light.",
  createdAt: "1687103690922",
  updatedAt: "1687103690923",
};

const mockAuthor = {
  id: "that-exact-user-id",
  name: "John Doe",
};

describe("Component PostPreview", () => {
  it("Should render as expected", () => {
    const { container } = render(
      <PostPreview post={mockPost} author={mockAuthor} />
    );
    expect(container).toMatchSnapshot();
  });
});
