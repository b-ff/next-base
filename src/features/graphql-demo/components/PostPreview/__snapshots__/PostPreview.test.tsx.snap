// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PostPreview Should render as expected 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column; gap: var(--spacing_2x);"
  >
    <div>
      <a
        href="/posts/1"
      >
        <h4>
          For whom the bell tolls?
           Â»
        </h4>
      </a>
    </div>
    <div>
      <small>
        Written by 
        John Doe
         at 
        18 June 2023 at 16:54
      </small>
    </div>
  </section>
</div>
`;
